### PR TITLE
Pages: Add template list

### PR DESCRIPTION
**Pull request by taipeicoder**
Related to #85686

## Proposed Changes

This PR adds the templates 404 and Search to the Calypso Pages page.  

## Testing Instructions

* Head to `/pages`.
* Ensure you see the templates Pages: 404 and Search Results at the bottom of the page.
* Ensure that clicking these templates open the Site Editor with the correct template selected.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/data-stores/src/templates/use-templates.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const useTemplates = (
+	siteId: string | number,
+	queryOptions: Omit< Options, 'queryKey' > = {}
+): UseQueryResult< Template[] > => {
+	return useQuery< Template[] >( {
+		queryKey: [ siteId, 'templates' ],
+		queryFn: () =>
+			wpcomRequest( {
+				path: `/sites/${ siteId }/templates`,
```
Comment (arthur791004) [inline packages/data-stores/src/templates/use-templates.ts:17]: Note that we have to make changes to support forwarding templates endpoint to the atomic sites.

Comment (taipeicoder) [reply packages/data-stores/src/templates/use-templates.ts:17]: Thanks for the reminder! I've added the endpoint to the API mapper via D133687-code.

**Comment by taipeicoder** [comment]
Moving this to the icebox.

**Comment by taipeicoder** [comment]
Closing due to the issue being stale, and no current plans to take further action on it.

**Pull request closed without merging**